RELEASES.md

```diff
@@ -0,0 +1,414 @@
+function.
+
+But now our constructor function is in the same place as the rest of our contract functions, within
+the general implementation of the storage struct.
+
+We tag these functions with the `#[ink(constructor)]` attribute. We can create multiple different
+constructors by simply creating more functions with the same tag. You can name a constructor
+function whatever you want (except starting with `__ink` which is reserved for all functions).
+
+<table style="width: 100%;">
+<tr>
+<th>Before</th>
+<th>After</th>
+</tr>
+<tr>
+<td>
+
+```rust
+impl Deploy for Erc20 {
+    fn deploy(&mut self, init_supply: Balance) {
+        let caller = env.caller();
+        self.total_supply.set(init_value);
+        self.balances.insert(caller, init_supply);
+        env.emit(Transfer {
+            from: None,
+            to: Some(env.caller()),
+            value: init_value
+        });
+    }
+}
+```
+
+</td>
+<td>
+
+```rust
+impl Erc20 {
+    #[ink(constructor)]
+    fn new(&mut self, initial_supply: Balance) {
+        let caller = self.env().caller();
+        self.total_supply.set(initial_supply);
+        self.balances.insert(caller, initial_supply);
+        self.env().emit_event(Transferred {
+            from: None,
+            to: Some(caller),
+            amount: initial_supply,
+        });
+    }
+}
+```
+
+</td>
+</tr>
+</table>
+
+## Cross Contract Calls
+
+It is now possible to call ink! messages and ink! constructors. So ink! constructors allow
+delegation and ink! messages can easily call other ink! messages.
+
+Given another ink! contract like `mod Accumulator { ... }`, we can call any of its functions just
+like we would expect:
+
+```rust
+use accumulator::Accumulator;
+let result = Accumulator::my_function()
+```
+
+## Factory Contracts
+
+Creation of other contracts from a factory contract works pretty much the same way it did in the old
+ink! language.
+
+However, users are now required to specify the `code_hash` separately rather than in the
+constructor:
+
+```rust
+.using_code(code_hash)
+```
+
+Also, they need to specify the used ink! environment (most likely `self.env()`):
+
+```rust
+create_using(self.env())
+```
+
+<table style="width: 100%;">
+<tr>
+<th>Before</th>
+<th>After</th>
+</tr>
+<tr>
+<td>
+
+```rust
+let accumulator = Accumulator::new(accumulator_code_hash, init_value)
+    .value(total_balance / 4)
+    .create()
+    .expect("failed at instantiating the accumulator contract");
+```
+
+</td>
+<td>
+
+```rust
+let accumulator = Accumulator::new(init_value)
+    .value(total_balance / 4)
+    .using_code(accumulator_code_hash)
+    .create_using(self.env())
+    .expect("failed at instantiating the `Accumulator` contract");
+```
+
+</td>
+</tr>
+</table>
+
+## Contract Tests
+
+Testing contracts off-chain is done by `cargo test` and users can simply use the standard routines
+of creating unit test modules within the ink! module itself:
```

Comment:
Users are free to declare their unit-tests wherever they want. Maybe we shall make that more clear at this point.

Reply:
I changed the line to "within the ink! project."